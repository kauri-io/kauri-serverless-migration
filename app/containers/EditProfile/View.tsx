import React, { Component } from 'react'
import styled from 'styled-components'
import EditProfile from '../EditProfileForm'
import PrimaryButton from '../../components/Button/PrimaryButton'
import Loading from '../../components/Loading'
import analytics from '../../lib/analytics'
import moment from 'moment'

const Page = styled.div`
    display: flex;
    background: ${props => props.theme.bgPrimary};
    height: 100vh;
    align-items: center;
    justify-content: center;
    flex-direction: column;
`

const Wrapper = styled.div`
    display: flex;
    flex-direction: column;
`

const ButtonWrapper = styled.div`
    display: flex;
    flex-direction: row;
    margin-top: ${props => props.theme.space[3]}px;
    justify-content: center;
    width: 100%;
`

class OnboardingEditProfile extends Component {
    handleSubmit() {
        const loginComp = this.login.getWrappedInstance().getWrappedInstance()

        const emailCheck = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
        const validEmail = emailCheck.test(
            String(loginComp.state.email).toLowerCase()
        )

        if (validEmail) {
            loginComp.saveUser(
                this.props.router.query.redirected
                    ? `${this.props.router.query.r}?redirected=true`
                    : this.props.router.query.r
            )
        } else {
            this.props.showNotificationAction({
                notificationType: 'error',
                message: 'Email Required',
                description: 'Please enter a valid email address',
            })
        }
    }

    componentDidMount() {
        const {
            name,
            username,
            email,
            avatar,
            social,
            title,
            website,
            dateCreated,
        } = this.props.user
        const github = social && social.github
        const twitter = social && social.twitter
        const hasData = name && username && email

        const loginTrackingPending = window.localStorage.getItem(
            'login-tracking-pending'
        )

        if (loginTrackingPending) {
            const daysCreated = moment().diff(moment(dateCreated), 'minutes')
            if (!daysCreated || daysCreated <= 5) {
                analytics.signup(this.props.user)
            } else {
                analytics.login(this.props.user)
            }
            window.localStorage.removeItem('login-tracking-pending')
        }
        if (hasData) {
            let newRedirectURL
            if (typeof this.props.router.query.r === 'string') {
                newRedirectURL =
                    this.props.router.query.r.indexOf('https://') !== -1 ||
                    this.props.router.query.redirected
                        ? this.props.router.query.r + '?redirected=true'
                        : this.props.router.query.r
            } else {
                newRedirectURL = `/public-profile/${this.props.userId}`
            }
            return this.props.routeChangeAction(newRedirectURL)
        }
    }

    render() {
        const {
            name,
            username,
            email,
            avatar,
            social,
            title,
            website,
        } = this.props.user
        const github = social && social.github
        const twitter = social && social.twitter
        const hasData = name && username && email
        if (hasData) {
            return (
                <Page>
                    <Loading />
                </Page>
            )
        }
        return (
            <Page>
                <Wrapper>
                    <EditProfile ref={comp => (this.login = comp)} />
                    <ButtonWrapper>
                        <PrimaryButton onClick={() => this.handleSubmit()}>
                            Next
                        </PrimaryButton>
                    </ButtonWrapper>
                </Wrapper>
            </Page>
        )
    }
}

export default OnboardingEditProfile
