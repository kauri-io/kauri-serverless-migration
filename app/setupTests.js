import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import theme from './lib/theme-config'
import { ThemeProvider } from 'styled-components'

Enzyme.configure({ adapter: new Adapter() })

const observeMock = {
    observe: () => null,
    disconnect: () => null, // maybe not needed
}

window.IntersectionObserver = () => observeMock

export const mountWithTheme = children =>
    mount(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const renderWithTheme = children =>
    render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const shallowWithTheme = children =>
    shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
