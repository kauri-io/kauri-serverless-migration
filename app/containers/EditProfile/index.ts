import View from './View'
import { connect } from 'react-redux'
import { routeChangeAction } from '../../lib/Epics/RouteChangeEpic'
import { showNotificationAction } from '../../lib/Epics/ShowNotificationEpic'
import { IReduxState } from '../../lib/Module'
import { WithRouterProps } from 'next/router'
interface IProps {
    router: WithRouterProps
}

const mapStateToProps = (state: IReduxState, ownProps: IProps) => ({
    userId: state.app.user && state.app.user.id,
    user: state.app.user,
})

export default connect(
    mapStateToProps,
    { routeChangeAction, showNotificationAction }
)(View)
