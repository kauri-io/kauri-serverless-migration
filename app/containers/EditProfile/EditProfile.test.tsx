import EditProfile from './View'
import { mountWithRedux } from '../../setupTests'
import { MockedProvider } from 'react-apollo/test-utils'

describe('containers/EditProfile', () => {
    it('snapshot should match', () => {
        const props = {
            showNotificationAction: jest.fn(),
            routeChangeAction: jest.fn(),
            router: {
                query: {
                    r: 'route',
                    redirected: true,
                },
            },
            user: {
                id: '123',
                name: 'test name',
                username: 'test userename',
                email: 'test@test.com',
                dateCreated: '2019-01-01',
            },
            userId: '123',
        }
        const wrapper = mountWithRedux(
            <MockedProvider>
                <EditProfile {...props} />
            </MockedProvider>
        )
        expect(wrapper).toMatchSnapshot()
    })
})
