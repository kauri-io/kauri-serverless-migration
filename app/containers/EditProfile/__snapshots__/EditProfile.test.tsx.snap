// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/EditProfile snapshot should match 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ThemeProvider
    theme={
      Object {
        "DEFAULT_CARD_WIDTH": 288,
        "MAIN_BODY_WIDTH": 1242,
        "bg": Object {
          "bgPrimary": "#1E2428",
          "bgSecondary": "#1E3D3B",
          "borderTextArea": "#979797",
          "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
          "disabledBackgroundColor": "#D6D6D6",
          "disabledTextColor": "#9B9B9B",
          "divider": "#EBEBEB",
          "errorRedColor": "#C03030",
          "hoverTextColor": "#108E72",
          "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
          "paddingTop": "2.5em",
          "primary": "#0BA986",
          "primaryColor": "#0BA986",
          "primaryDark": "#11856B",
          "primaryTextColor": "#1E2428",
          "secondary": "#11856B",
          "secondaryBlue": "rgb(81, 123, 162, 1)",
          "secondaryBlueDark": "rgb(81, 118, 142, 1)",
          "secondaryColor": "#1E3D3B",
          "secondaryTextColor": "#283035",
          "tertiaryBackgroundColor": "#F7F7F7",
          "textPrimary": "#1E2428",
          "white": "#FFFFFF",
        },
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "breakpoints": Array [
          "500px",
          "52em",
          "1242px",
        ],
        "colors": Object {
          "bgPrimary": "#1E2428",
          "bgSecondary": "#1E3D3B",
          "borderTextArea": "#979797",
          "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
          "disabledBackgroundColor": "#D6D6D6",
          "disabledTextColor": "#9B9B9B",
          "divider": "#EBEBEB",
          "errorRedColor": "#C03030",
          "hoverTextColor": "#108E72",
          "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
          "paddingTop": "2.5em",
          "primary": "#0BA986",
          "primaryColor": "#0BA986",
          "primaryDark": "#11856B",
          "primaryTextColor": "#1E2428",
          "secondary": "#11856B",
          "secondaryBlue": "rgb(81, 123, 162, 1)",
          "secondaryBlueDark": "rgb(81, 118, 142, 1)",
          "secondaryColor": "#1E3D3B",
          "secondaryTextColor": "#283035",
          "tertiaryBackgroundColor": "#F7F7F7",
          "textPrimary": "#1E2428",
          "white": "#FFFFFF",
        },
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "fontSizes": Array [
          11,
          13,
          14,
          16,
          18,
          20,
          22,
          28,
          32,
        ],
        "fontWeight": Array [
          300,
          "normal",
          500,
          700,
        ],
        "hoverTextColor": "#108E72",
        "padContent": "
    max-width: 1242px;
    margin-left: auto;
    margin-right: auto;
    @media (min-width: 1242px) {
      min-width: 1242px;
    }
    @media (max-width: 1242px) {
      min-width: 90%;
    }
  ",
        "padding": "calc((100vw - 1242px) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "space": Array [
          0,
          10,
          15,
          30,
          60,
        ],
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      }
    }
  >
    <MockedProvider
      addTypename={true}
    >
      <ApolloProvider
        client={
          ApolloClient {
            "__operations_cache__": Map {},
            "cache": InMemoryCache {
              "addTypename": true,
              "cacheKeyRoot": KeyTrie {
                "weakness": true,
              },
              "config": Object {
                "addTypename": true,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "freezeResults": false,
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
                "executeSubSelectedArray": [Function],
                "freezeResults": false,
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
            "clearStoreCallbacks": Array [],
            "defaultOptions": Object {},
            "disableNetworkFetches": false,
            "link": MockLink {
              "addTypename": true,
              "mockedResponsesByKey": Object {},
            },
            "localState": LocalState {
              "cache": InMemoryCache {
                "addTypename": true,
                "cacheKeyRoot": KeyTrie {
                  "weakness": true,
                },
                "config": Object {
                  "addTypename": true,
                  "dataIdFromObject": [Function],
                  "fragmentMatcher": HeuristicFragmentMatcher {},
                  "freezeResults": false,
                  "resultCaching": true,
                },
                "data": DepTrackingCache {
                  "data": Object {},
                  "depend": [Function],
                },
                "maybeBroadcastWatch": [Function],
                "optimisticData": DepTrackingCache {
                  "data": Object {},
                  "depend": [Function],
                },
                "silenceBroadcast": false,
                "storeReader": StoreReader {
                  "executeSelectionSet": [Function],
                  "executeStoreQuery": [Function],
                  "executeSubSelectedArray": [Function],
                  "freezeResults": false,
                },
                "storeWriter": StoreWriter {},
                "typenameDocumentCache": Map {},
                "watches": Set {},
              },
              "client": [Circular],
            },
            "mutate": [Function],
            "query": [Function],
            "queryDeduplication": true,
            "queryManager": QueryManager {
              "assumeImmutableResults": false,
              "clientAwareness": Object {
                "name": undefined,
                "version": undefined,
              },
              "dataStore": DataStore {
                "cache": InMemoryCache {
                  "addTypename": true,
                  "cacheKeyRoot": KeyTrie {
                    "weakness": true,
                  },
                  "config": Object {
                    "addTypename": true,
                    "dataIdFromObject": [Function],
                    "fragmentMatcher": HeuristicFragmentMatcher {},
                    "freezeResults": false,
                    "resultCaching": true,
                  },
                  "data": DepTrackingCache {
                    "data": Object {},
                    "depend": [Function],
                  },
                  "maybeBroadcastWatch": [Function],
                  "optimisticData": DepTrackingCache {
                    "data": Object {},
                    "depend": [Function],
                  },
                  "silenceBroadcast": false,
                  "storeReader": StoreReader {
                    "executeSelectionSet": [Function],
                    "executeStoreQuery": [Function],
                    "executeSubSelectedArray": [Function],
                    "freezeResults": false,
                  },
                  "storeWriter": StoreWriter {},
                  "typenameDocumentCache": Map {},
                  "watches": Set {},
                },
              },
              "fetchQueryRejectFns": Map {},
              "idCounter": 1,
              "inFlightLinkObservables": Map {},
              "link": MockLink {
                "addTypename": true,
                "mockedResponsesByKey": Object {},
              },
              "localState": LocalState {
                "cache": InMemoryCache {
                  "addTypename": true,
                  "cacheKeyRoot": KeyTrie {
                    "weakness": true,
                  },
                  "config": Object {
                    "addTypename": true,
                    "dataIdFromObject": [Function],
                    "fragmentMatcher": HeuristicFragmentMatcher {},
                    "freezeResults": false,
                    "resultCaching": true,
                  },
                  "data": DepTrackingCache {
                    "data": Object {},
                    "depend": [Function],
                  },
                  "maybeBroadcastWatch": [Function],
                  "optimisticData": DepTrackingCache {
                    "data": Object {},
                    "depend": [Function],
                  },
                  "silenceBroadcast": false,
                  "storeReader": StoreReader {
                    "executeSelectionSet": [Function],
                    "executeStoreQuery": [Function],
                    "executeSubSelectedArray": [Function],
                    "freezeResults": false,
                  },
                  "storeWriter": StoreWriter {},
                  "typenameDocumentCache": Map {},
                  "watches": Set {},
                },
                "client": [Circular],
              },
              "mutationStore": MutationStore {
                "store": Object {},
              },
              "onBroadcast": [Function],
              "pollingInfoByQueryId": Map {},
              "queries": Map {},
              "queryDeduplication": true,
              "queryStore": QueryStore {
                "store": Object {},
              },
              "ssrMode": false,
              "transformCache": WeakMap {},
            },
            "reFetchObservableQueries": [Function],
            "resetStore": [Function],
            "resetStoreCallbacks": Array [],
            "store": DataStore {
              "cache": InMemoryCache {
                "addTypename": true,
                "cacheKeyRoot": KeyTrie {
                  "weakness": true,
                },
                "config": Object {
                  "addTypename": true,
                  "dataIdFromObject": [Function],
                  "fragmentMatcher": HeuristicFragmentMatcher {},
                  "freezeResults": false,
                  "resultCaching": true,
                },
                "data": DepTrackingCache {
                  "data": Object {},
                  "depend": [Function],
                },
                "maybeBroadcastWatch": [Function],
                "optimisticData": DepTrackingCache {
                  "data": Object {},
                  "depend": [Function],
                },
                "silenceBroadcast": false,
                "storeReader": StoreReader {
                  "executeSelectionSet": [Function],
                  "executeStoreQuery": [Function],
                  "executeSubSelectedArray": [Function],
                  "freezeResults": false,
                },
                "storeWriter": StoreWriter {},
                "typenameDocumentCache": Map {},
                "watches": Set {},
              },
            },
            "typeDefs": undefined,
            "version": "2.6.3",
            "watchQuery": [Function],
          }
        }
      >
        <OnboardingEditProfile
          routeChangeAction={
            [MockFunction] {
              "calls": Array [
                Array [
                  "route?redirected=true",
                ],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            }
          }
          router={
            Object {
              "query": Object {
                "r": "route",
                "redirected": true,
              },
            }
          }
          showNotificationAction={[MockFunction]}
          user={
            Object {
              "dateCreated": "2019-01-01",
              "email": "test@test.com",
              "id": "123",
              "name": "test name",
              "username": "test userename",
            }
          }
          userId="123"
        >
          <View__Page>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "View__Page-sc-8ju51k-0",
                    "isStatic": false,
                    "lastClassName": "eFXWaF",
                    "rules": Array [
                      "display:flex;background:",
                      [Function],
                      ";height:100vh;align-items:center;justify-content:center;flex-direction:column;",
                    ],
                  },
                  "displayName": "View__Page",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "View__Page-sc-8ju51k-0",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="View__Page-sc-8ju51k-0 eFXWaF"
              >
                <LoadingComponent>
                  <style
                    type="text/css"
                  >
                    
        .centered-wrapper{
          display: flex;
          flex: 1;
          align-items: center;
          justify-content: center;
        }
        .lds-ellipsis {
          display: inline-block;
          position: relative;
          width: 64px;
          height: 64px;
        }
        .lds-ellipsis div {
          position: absolute;
          top: 27px;
          width: 11px;
          height: 11px;
          border-radius: 50%;
          background: #0BA986;
          animation-timing-function: cubic-bezier(0, 1, 1, 0);
        }
        .lds-ellipsis div:nth-child(1) {
          left: 6px;
          animation: lds-ellipsis1 0.6s infinite;
        }
        .lds-ellipsis div:nth-child(2) {
          left: 6px;
          animation: lds-ellipsis2 0.6s infinite;
        }
        .lds-ellipsis div:nth-child(3) {
          left: 26px;
          animation: lds-ellipsis2 0.6s infinite;
        }
        .lds-ellipsis div:nth-child(4) {
          left: 45px;
          animation: lds-ellipsis3 0.6s infinite;
        }
        @keyframes lds-ellipsis1 {
          0% {
            transform: scale(0);
          }
          100% {
            transform: scale(1);
          }
        }
        @keyframes lds-ellipsis3 {
          0% {
            transform: scale(1);
          }
          100% {
            transform: scale(0);
          }
        }
        @keyframes lds-ellipsis2 {
          0% {
            transform: translate(0, 0);
          }
          100% {
            transform: translate(19px, 0);
          }
        }
        
                  </style>
                  <div
                    className="centered-wrapper"
                  >
                    <div
                      className="lds-ellipsis"
                    >
                      <div />
                      <div />
                      <div />
                      <div />
                    </div>
                  </div>
                </LoadingComponent>
              </div>
            </StyledComponent>
          </View__Page>
        </OnboardingEditProfile>
      </ApolloProvider>
    </MockedProvider>
  </ThemeProvider>
</Provider>
`;
